pkg/cmd/roachtest/tests/backup.go

```diff
@@ -1103,6 +1104,368 @@ func registerBackup(r registry.Registry) {
+		`'gs://cockroach-fixtures/tpch-csv/sf-100/orders.tbl.3?AUTH=implicit'`,
+		`'gs://cockroach-fixtures/tpch-csv/sf-100/orders.tbl.5?AUTH=implicit'`,
+		`'gs://cockroach-fixtures/tpch-csv/sf-100/orders.tbl.7?AUTH=implicit'`,
+	}
+	var jobID string
+	err = conn.QueryRowContext(ctx, fmt.Sprintf(
+		`IMPORT INTO orders CSV DATA (%s) WITH delimiter='|', detached`, strings.Join(files, ", ")),
+	).Scan(&jobID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	waitForStatus(jobID, jobs.StatusSucceeded, "", 20*time.Minute)
+
+	// Take a backup of the database, and fingerprint it for restore comparison.
+	// We do a full database backup in order to do a final incremental backup
+	// after the table is dropped.
+	t.Status("fingerprinting at time 'initial'")
+	timeInitial := now()
+	fpInitial, err := fingerprint(ctx, conn, "tpch", "orders")
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Status("fingerprint:\n", fpInitial)
+
+	t.Status("taking full backup")
+	dest := fmt.Sprintf("nodelocal://1/%s", destinationName(c))
+	err = conn.QueryRowContext(ctx,
+		`BACKUP DATABASE tpch INTO $1 WITH revision_history, detached`, dest).Scan(&jobID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	waitForStatus(jobID, jobs.StatusSucceeded, "", 10*time.Minute)
+
+	// Import and cancel even-numbered files twice.
+	files = []string{
+		`'gs://cockroach-fixtures/tpch-csv/sf-100/orders.tbl.2?AUTH=implicit'`,
+		`'gs://cockroach-fixtures/tpch-csv/sf-100/orders.tbl.4?AUTH=implicit'`,
+		`'gs://cockroach-fixtures/tpch-csv/sf-100/orders.tbl.6?AUTH=implicit'`,
+		`'gs://cockroach-fixtures/tpch-csv/sf-100/orders.tbl.8?AUTH=implicit'`,
+	}
+	for i := 0; i < 2; i++ {
+		t.Status("importing even-numbered files")
+		_, err = conn.ExecContext(ctx,
+			`SET CLUSTER SETTING jobs.debug.pausepoints = 'import.after_ingest'`)
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = conn.QueryRowContext(ctx, fmt.Sprintf(
+			`IMPORT INTO orders CSV DATA (%s) WITH delimiter='|', detached`, strings.Join(files, ", ")),
+		).Scan(&jobID)
+		if err != nil {
+			t.Fatal(err)
+		}
+		waitForStatus(jobID, jobs.StatusPaused, "", 20*time.Minute)
+
+		t.Status("canceling import")
+		_, err = conn.ExecContext(ctx, fmt.Sprintf(`CANCEL JOB %s`, jobID))
+		if err != nil {
+			t.Fatal(err)
+		}
+		waitForStatus(jobID, jobs.StatusCanceled, "", 20*time.Minute)
+	}
+
+	// Check that we actually wrote MVCC range tombstones.
+	var rangeKeys int
+	err = conn.QueryRowContext(ctx, `
+		SELECT SUM((crdb_internal.range_stats(start_key)->'range_key_count')::INT)
+		FROM crdb_internal.ranges
+		WHERE database_name = 'tpch' AND table_name = 'orders'
+	`).Scan(&rangeKeys)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if rangeKeys == 0 {
+		t.Fatal("no MVCC range tombstones found")
+	}
+
+	// Fingerprint for restore comparison.
+	t.Status("fingerprinting at time 'canceled'")
+	timeCanceled := now()
+	fpCanceled, err := fingerprint(ctx, conn, "tpch", "orders")
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
nit: use `require.NoError(t,err)` ?

Reply:
Wasn't sure if that was kosher in roachtests? Definitely prefer that though. Seems like it's used in quite a few spots, will update tomorrow.